pkg/controller/podautoscaler/metrics/rest_metrics_client.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clientv1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	autoscaling "k8s.io/kubernetes/pkg/apis/autoscaling/v2alpha1"
+	customapi "k8s.io/metrics/pkg/apis/custom_metrics/v1alpha1"
+	resourceclient "k8s.io/metrics/pkg/client/clientset_generated/clientset/typed/metrics/v1alpha1"
+	customclient "k8s.io/metrics/pkg/client/custom_metrics"
+)
+
+func NewRESTMetricsClient(resourceClient resourceclient.PodMetricsesGetter, customClient customclient.CustomMetricsClient) MetricsClient {
+	return &restMetricsClient{
+		&resourceMetricsClient{resourceClient},
+		&customMetricsClient{customClient},
+	}
+}
+
+// FullMetricsClient is a client which supports fetching
+// metrics from both the resource metrics API and the
+// custom metrics API.
+type restMetricsClient struct {
+	*resourceMetricsClient
+	*customMetricsClient
+}
+
+// resourceMetricsClient implements the resource-metrics-related parts of MetricsClient,
+// using data from the reosurce metrics API.
+type resourceMetricsClient struct {
+	client resourceclient.PodMetricsesGetter
+}
+
+// GetResourceMetric gets the given resource metric (and an associated oldest timestamp)
+// for all pods matching the specified selector in the given namespace
+func (c *resourceMetricsClient) GetResourceMetric(resource v1.ResourceName, namespace string, selector labels.Selector) (PodMetricsInfo, time.Time, error) {
+	metrics, err := c.client.PodMetricses(namespace).List(metav1.ListOptions{LabelSelector: selector.String()})
```

Comment:
I'll post a follow up PR -- it needs changes to gengo to detect already-plural type names and not attempt to pluralize them.